src/cryptography/hazmat/primitives/kdf/concatkdf.py

```diff
@@ -0,0 +1,94 @@
+# This file is dual licensed under the terms of the Apache License, Version
+# 2.0, and the BSD License. See the LICENSE file in the root of this repository
+# for complete details.
+
+from __future__ import absolute_import, division, print_function
+
+import struct
+
+from cryptography import utils
+from cryptography.exceptions import (
+    AlreadyFinalized, InvalidKey, UnsupportedAlgorithm, _Reasons
+)
+from cryptography.hazmat.backends.interfaces import HMACBackend
+from cryptography.hazmat.primitives import constant_time, hashes, hmac
+from cryptography.hazmat.primitives.kdf import KeyDerivationFunction
+
+
+@utils.register_interface(KeyDerivationFunction)
+class ConcatKDF(object):
+    def __init__(self, algorithm, length, salt, otherinfo, usehmac, backend):
+        if not isinstance(backend, HMACBackend) and usehmac is True:
+            raise UnsupportedAlgorithm(
+                "Backend object does not implement HMACBackend.",
+                _Reasons.BACKEND_MISSING_INTERFACE
+            )
+
+        self._algorithm = algorithm
+
+        max_length = algorithm.digest_size * (2 ** 32 - 1)
+        if length > max_length:
+            raise ValueError(
+                "Can not derive keys larger than {0} bits.".format(
+                    max_length
+                ))
+        self._length = length
+
+        if not isinstance(salt, bytes) and salt is not None:
```

Comment:
Is the `salt is not None` bit necessary? `isinstance(None, bytes)` is always False.